hr_expense_analytic_tag/models/account_move_line.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _prepare_analytic_lines(self):
+        vals = super()._prepare_analytic_lines()
```

Comment:
```suggestion
        vals_list = super()._prepare_analytic_lines()
```

Reply:
Anyway, this is not the place to do this (apart from the current bad algorithm to match the line). You should put the analytic tags on the `account.move.line`, not on the analytic line, and this will be done automatically.

Reply:
Ok, changed to a more appropriate approach.